docs/chapters/ci-and-cd/setting-up-automated-testing-yocto.rst

```diff
@@ -0,0 +1,176 @@
+partition #4.
+
+.. code-block:: bash
+
+    sudo fdisk /dev/sd<YOUR USB DEVICE>
+    - Press n for new partition
+    - Press p for a primary partition
+    - Press enter twice to get it to fill all remaining space
+    - Press w to write
+    sudo mkfs.ext3 -L testrootfs /dev/sd<YOUR USB DEVICE>4
+
+Now mount the testmaster rootfs in order to do some changes there:
+
+.. code-block:: bash
+
+    sudo mount /dev/sd<YOUR USB DEVICE>2 /mnt/
+      
+    # Create image used to identify if system is booted into the testmaster image
+    sudo touch /mnt/etc/masterimage
+      
+    # Make sure the testmaster image shows a login prompt
+    sudo ln -sf /lib/systemd/system/getty@.service /mnt/etc/systemd/system/getty.target.wants/getty@tty1.service
+      
+    # Create a network conf file which we then copy to rootfs, this one uses DHCP
+    cat <<EOF > /tmp/20-wired.network
+    [Match]
+    Name=en*
+     
+    [Network]
+    DHCP=ipv4
+     
+    [DHCP]
+    RouteMetric=10
+    ClientIdentifier=mac
+    EOF
+    sudo cp /tmp/20-wired.network /mnt/etc/systemd/network/
+     
+      
+    # Unmount
+    sudo umount /mnt
+
+Now we mount the EFI partition to add a bootloader entry called "test" which boots the kernel and filesystem under test.
+
+.. code-block:: bash
+
+    # Set correct label on EFI partition
+    sudo dosfslabel /dev/sd<YOUR USB DEVICE>1 boot
+      
+    # Mount EFI partition
+    sudo mount /dev/sd<YOUR USB DEVICE>1 /mnt/
+    # Create temp bootloader config file which we then copy
+    cat <<EOF > /tmp/test.conf
+    title test
+    linux /test-kernel
+    options LABEL=test root=/dev/sda4 ro rootwait console=ttyS0 console=tty0
+    EOF
+    sudo cp /tmp/test.conf /mnt/loader/entries/
+      
+    # Unmount
+    sudo umount /mnt
+
+The USB-stick should now be ready and can be inserted into a NUC and booted, do that and check what IP-address it gets
+using e.g. "ip a".
+
+
+Building and testing an image
+-----------------------------
+Current issues in poky means we have to apply a patch needs to be applied: 
+`0001-Fix-automated-runtime-testing-using-SystemdbootTarge.patch` which is located in _TODO:_ Add link!
+
+.. code-block:: bash
+
+    cd ../sources/poky/
+    git am /path/to/0001-Fix-automated-runtime-testing-using-SystemdbootTarge.patch
+
+There is some configuration that needs to be setup in local.conf in order to enable target testing, so add the
+following to conf/local.conf
+
+.. code-block:: bash
+
+    IMAGE_FSTYPES += "tar.gz"
+    INHERIT += "testimage"
+    TEST_TARGET = "SystemdbootTarget"
+    TEST_TARGET_IP = "<IP of NUC>"
+    TEST_SERVER_IP = "<IP of machine used for building>"
+
+Sometimes we need to set TEST_SERVER_IP, although that shouldn't be neccesary according to the docs.
+This might be related to multiple network interfaces confusing the autodetection.
+
+You can now build and test basically any image using "bitbake -c testimage <my image>", e.g.:
```

Comment:
If it's possible to mark the bitbake command and args as inline code, please do so.